### DISCUSSION TITLE
How to style self-hosted tilemap using deck.gl

### DISCUSSION THREAD

**Discussion by VladMonk** [visgl/deck.gl#8069] [Q&A] [answered]
Hello there! I have .mbtiles map file from here - [bbbike](https://extract.bbbike.org/). Then i serve it using [tileserver-gl](http://tileserver.org/). In application i'm use MVTLayer to display it into DeckGL component(react). Map is displayed but without any styles(obviously). 

The question is how to apply styles? Prop "mapStyle", in wich i placed json object with styles, doesn't change anything. A'm sure that styles are correct, because in tileserver interface all they works fine. 

Next thing i'm sure, that current style import isn't right. I just copied style.json from directory with files(maps/styles/fonts/etc) for Tileserver. So, probably, i should get my styles different way, but how? And how to switch between them after?

It should be easy, but i cant find solution. Hope, someone help me)

`import json from './assets/style.json'`

`const initViewState = {
        longitude: 30.185,
        latitude: 59.9505,
        zoom: 11,
        minZoom: 5,
        maxZoom: 15,
    }`

`const tileLayer = new MVTLayer({
        data: "http://localhost:8080/data/openmaptiles/{z}/{x}/{y}.pbf",
        minZoom: initViewState.minZoom,
        maxZoom: initViewState.maxZoom
    })`

`<DeckGL
        initialViewState={spbInitViewState}
        controller={true}
        layers={[tileLayer]}
        mapStyle={json}
    />`

**Comment by Pessimistress** [comment, marked as answer]
MVTLayer is not a replacement of Mapbox. If you want to use Mapbox JSON styles, see [using with base map](https://deck.gl/docs/get-started/using-with-map).

Props supported by MVTLayer can be found on its [documentation](https://deck.gl/docs/api-reference/geo-layers/mvt-layer) page.
